mod_reforged/config/sling_items.nut

```diff
@@ -0,0 +1,96 @@
+// This class combines common properties of the new sling-an-item skills
+::Reforged.Skills.adjustSlingItemSkill <- function( o )
+{
+	o.m.ID = ::MSU.String.replace(o.m.ID, "throw", "sling");
+	o.m.Name = ::MSU.String.replace(o.m.Name, "Throw", "Sling");
+	o.m.Description = ::MSU.String.replace(::MSU.String.replace(o.m.Description, "throw", "sling"), "Throw", "Sling");
+	o.m.Order = ::Const.SkillOrder.UtilityTargeted + 1000;
+	o.m.ActionPointCost = 5;	// Flasks have 5
+	o.m.FatigueCost = 25;		// Flasks have 20
+	o.m.MinRange = 2;			// Flasks have 0
+	o.m.MaxRange = 6;			// Flasks have 3
+	o.m.MaxLevelDifference = 4;	// Flasks have 3
+	o.m.IsRanged = true;	// Causes you to be able to sling further when shooting downhill
+	o.m.IsHidden = true;
+
+	local getTooltip = o.getTooltip;
+	o.getTooltip = function()
+	{
+		local ret = getTooltip();
+		ret.push({
+			id = 8,
+			type = "text",
+			icon = "ui/icons/vision.png",
+			text = "Has a range of " + ::MSU.Text.colorGreen(this.getMaxRange()) + " tiles on even ground, more if slinging downhill"
+		});
+		if (this.getContainer().getActor().isEngagedInMelee() == false)
+		{
+			ret.push({
+				id = 9,
+				type = "text",
+				icon = "ui/tooltips/warning.png",
+				text = ::MSU.Text.colorRed("Can not be used because this character is engaged in melee")
+			});
+		}
+		return ret;
+	}
+
+	// We only show these skills while the respective items are in one of the bag slots
+	local isHidden = ::mods_getMember(o, "isHidden");
+	::mods_override(o, "isHidden", function() {
+		local mainhandItem = this.getContainer().getActor().getMainhandItem();
+		if (mainhandItem == null) return isHidden();
```

Comment:
In my opinion all of these should be in a single if block with ||. Discussion on this is already happening on a separate PR so let's go with whatever conclusion comes up there.